lib/reek/smells/manual_dispatch.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+
+module Reek
+  module Smells
+    #
+    # A Manual Dispatch occurs when a method is only called after a
+    # manual check that the method receiver is of the correct type.
+    #
+    # The +ManualDispatch+ checker reports any invocation of +respond_to?+
+    #
+    # See {file:docs/Manual-Dispatch.md} for details.
+    class ManualDispatch < SmellDetector
+      MSG = 'manually dispatches method call'.freeze
```

Comment:
We have the convention in our codebase to spell things out (with the exception of "ctx" and "exp") so let's rather go with MESSAGE